### ISSUE TITLE
Fona 3g

### ISSUE THREAD

**Issue by oaf2015** [climberhunt/PiPhone#2]
After running the piphone.py file, I could see the keypad but nothing happens on putty after main loop. Line 312 to be exact. Any recommendations?

**Comment by joeblankenship1** [comment]
Did the calibration test work for the touch screen?

**Comment by oaf2015** [comment]
Yes it did. I am able to put in numbers, delete, and even hit call but
nothing happens. On the terminal while logged in ssh I see an error right
at the calling line on mainloop.

**Comment by oaf2015** [comment]
I am wondering if it has anything to do with AT commands and having to
preconfigure the fona 3G prior to finalizing it with raspberry pi 3.

**Comment by climberhunt** [comment]
use minicom to talk to the FONA, and issue a few AT commands, try dialling a number "ATDT..." etc.. 
The SIM may be blocked, you might need to enter a PIN code. There's an AT code for that, I think. I seem to recall having to do that for one of my SIMs but nor all of them.

**Comment by oaf2015** [comment]
Sim is not locked. ATD plus number. Do you have an example of the complete
calling command. Minicom talks fine with fona no issues there

**Comment by climberhunt** [comment]
I'm afraid I haven't had a functioning PiPhone here for the last couple of years, my Fona module is in my front gates to open/close them, so I can't test it....

**Comment by oaf2015** [comment]
So don't even try it?

**Comment by joeblankenship1** [comment]
@oaf2015 I still have my PiPhone built, but I'm currently slammed with work projects. Did you need this test ran for work or for hobby? If it's work related, I'll try to get to it this week.

**Comment by oaf2015** [comment]
Work.

**Comment by oaf2015** [comment]
I really appreciate your help. I have gone through and attempted to
identify AT commands comparisons but found none. I did have to change the
serial.port to ttyS0 and it is communicating good. My sticking point right
now is the fact that it is incapable of delivering the call or receive
calls. On putty I see an error when I attempt to make a call. On the device
(pitft) it says calling....

**Comment by joeblankenship1** [comment]
I will test my PiPhone ASAP to validate your issues. I don't think I was able to receive calls; I was only ever able to dial out. I just hope I still have minutes on my SIM.

**Comment by oaf2015** [comment]
Thank you.

**Comment by joeblankenship1** [comment]
@oaf2015 I haven't forgot about you; I've just been really busy. Hopefully, you've already found a solution.

**Comment by oaf2015** [comment]
No, not yet. I appreciate. I am getting back on it Monday.

**Comment by Ericka430** [comment]
I'm trying to do this project for my senior project class and i keep getting an error that says List index out of range, what can I do to fix this?

**Comment by Ciemaar** [comment]
Can you show more of the error message?

**Comment by Ericka430** [comment]
![fullsizerender](https://cloud.githubusercontent.com/assets/27842356/25277757/0261a6aa-266e-11e7-83bb-c51b8519f2c3.jpg)

The last line is line 255

![img_1674](https://cloud.githubusercontent.com/assets/27842356/25277786/1c3eed4e-266e-11e7-9622-d465149679c7.JPG)
 
This is the error message that I get, (IndexError: list out of range)if possible could you please help me out.

**Comment by rimomaguiar** [comment]
In my case, I am using the sim900 gsm but I got stuck on the piphone splash logo, the keyboard never appears, any help?

**Comment by rimomcosta** [comment]
I was having this same error message when I was using an external monitor, I had to connect the pitft 2.8" monitor and call the piphone.py from the small screen and the program started.

**Comment by oaf2015** [comment]
I could never get it working. Moved on to a different project and gave up
on this.

**Comment by Mtaganov** [comment]
If you get the program working, (keypads on) but when you dial, nothing happens, my solution: 
I used "USB to UART" cable to talk to fona. But in pyphone.py code, "serialport = serial.Serial("/dev/ttyAMA0", 115200, timeout=0.5)" this line, I changed ttyAMA0 to ttyUSB0, and the program was able to talk.
